src/app/admin/group/group.component.ts

```diff
@@ -0,0 +1,274 @@
+import { Component, OnInit, ViewChild, Inject } from "@angular/core";
+import { HttpService } from "../../shared/http.service";
+import { Group, Speciality, Faculty } from "./entity.interface";
+import { MatTableDataSource, MatTable } from "@angular/material";
+import { MatPaginator } from "@angular/material/paginator";
+import {
+  MatDialog,
+  MatDialogRef,
+  MAT_DIALOG_DATA
+} from "@angular/material/dialog";
+
+export interface DialogData {
+  data: any;
+}
+
+@Component({
+  selector: "app-group",
+  templateUrl: "./group.component.html",
+  styleUrls: ["./group.component.scss"],
+  providers: [HttpService, MatDialog]
+})
+export class GroupComponent implements OnInit {
+  listGroups: Group[] = [];
+  dataSource = new MatTableDataSource<Group>();
+  displayedColumns: string[] = [
+    "id",
+    "name",
+    "but_student",
+    "but_edit",
+    "but_del"
+  ];
+
+  @ViewChild("table", { static: true }) table: MatTable<Group>;
+  @ViewChild(MatPaginator, { static: true }) paginator: MatPaginator;
+
+  applyFilter(filterValue: string) {
+    this.dataSource.filter = filterValue.trim().toLowerCase();
+  }
+
+  constructor(protected httpService: HttpService, public dialog: MatDialog) {}
+
+  ngOnInit() {
+    this.viewAllGroups();
+  }
+
+  viewAllGroups() {
+    this.httpService.getRecords("group").subscribe((result: Group[]) => {
+      this.listGroups = result;
+      this.dataSource.data = this.listGroups;
+      console.log(result);
+    });
+    this.dataSource.paginator = this.paginator;
+  }
+
+
+  // add modal window for add new group
+  addGroupDialog(group: Group): void {
+    const dialogRef = this.dialog.open(GroupComponentAdd, {
+      width: "500px",
+      data: {}
+    });
+
+    dialogRef.afterClosed().subscribe(result => {
+      console.log("The dialog was closed");
+      console.log(result);
+      if (result) {
+        this.addGroup(result);
+      }
+    });
+  }
+  /** Add new group*/
+  addGroup(group: Group) {
+    this.httpService.insertData("group", group).subscribe((result: Group[]) => {
+      this.listGroups.push(result[0]);
```

Comment:
Can we avoid forcing table render by creating new instance for listGroups?

Reply:
Result contains only new group. When create new instance for listGroups with result, output table has one row

Reply:
this.listGroups = [ ...this.listGroups, result[0] ]